conftest.py

```diff
@@ -111,156 +276,87 @@ def temp_dir():
-    answers. If an answer file already exists, then it cannot be used
-    to hold newly generated answers; a fresh file must be used instead.
-
-    Each class that performs answer tests (e.g., TestEnzo) has a
-    corresponding answer file. These answer files are stored in
-    test_data_dir/answers. The names of the files and their
-    corresponding answer class are in ./tests/tests.yaml.
-
-    Parameters:
-    -----------
-        request : pytest.FixtureRequest
-            Provides access to the requesting test context. For
-            example, if an answer class uses this fixture, such as
-            TestEnzo, then request provides access to all of the
-            methods and attributes of the TestEnzo class, since
-            that class is the user of this fixture (the calling
-            context).
-
-    Example:
-    --------
-        # This fixture should be used whenever a new answer class is
-        # defined
-        >>> @pytest.mark.usefixtures('answer_file')
-        >>> class TestNewFrontend:
-        >>>     def test1(self):
-                    ...
+def ds(request):
+    try:
+        if isinstance(request.param, list):
+            # data_dir_load can take the cls, args, and kwargs. These optional
+            # arguments are given in a dictionary as the second element of the
+            # list
+            ds_fn = request.param[0]
+            opts = request.param[1]
+            cls = None if "cls" not in opts.keys() else opts["cls"]
+            args = None if "args" not in opts.keys() else opts["args"]
+            kwargs = None if "kwargs" not in opts.keys() else opts["kwargs"]
+            dataset = data_dir_load(ds_fn, cls=cls, args=args, kwargs=kwargs)
+        else:
+            dataset = data_dir_load(request.param)
+        return dataset
+    except FileNotFoundError:
+        return pytest.skip(f"Data file: `{request.param}` not found.")
+
+
+@pytest.fixture(scope="class")
+def f(request):
     """
-    if request.cls.__name__ in answer_files:
-        answer_file = answer_files[request.cls.__name__]
-        # Make sure we're not overwriting an existing answer set
-        if os.path.isfile(os.path.join(answer_dir, answer_file)):
-            if request.config.getoption("--answer-store"):
-                raise FileExistsError(
-                    "Error, attempting to overwrite "
-                    "answer file {}. Either specify a new version or "
-                    "set the `--force-override-answers` option".format(answer_file)
-                )
-    else:
-        raise AssertionError
-    request.cls.answer_file = answer_file
+    Fixture for returning the field. Needed because indirect=True is
+    used for loading the datasets.
+    """
+    return request.param
 
 
-def _param_list(request):
-    r"""
-    Saves the non-ds, non-fixture function arguments for saving to
-    the answer file.
+@pytest.fixture(scope="class")
+def d(request):
```

Comment:
It would be much better to have something more explicit than just `d` (or `a`, `w`,`N`).

Reply:
I also don't quite understand how this is necessary from the docstring?

Reply:
See the comment I left below on the change to `a`

Reply:
I changed all of the offending fixture names. I'll have to change them in the tests as they are added in future PRs.